src/main/java/life/qbic/subscriptions/SubscriptionController.java

```diff
@@ -45,13 +48,38 @@
   SubscriptionController(
       SubscriptionRepository subscriptionRepository,
       RequestDecrypter requestDecrypter,
-      RequestEncrypter requestEncrypter
-  ) {
+      RequestEncrypter requestEncrypter) {
     this.subscriptionRepository = subscriptionRepository;
     this.requestDecrypter = requestDecrypter;
     this.requestEncrypter = requestEncrypter;
   }
 
+  @Operation(summary = "Cancel a subscription",
+      parameters = {
+          @Parameter(name = "token",
+              description = "The token of an encrypted cancel request.",
+              example = "For_lfbnS9iTi4Nmwnei4LA_f8SHga1Rdz4yw6aT8zz0V8PaHm1QEbKQTv1jGCEA", schema = @Schema(implementation = String.class))
+      },
+      responses = {
+          @ApiResponse(responseCode = "204", description = "No content. Subscription cancelled, token accepted.",
+              content = @Content(mediaType = "application/json",
+                  schema = @Schema(implementation = CancellationRequest.class))
+          ),
+          @ApiResponse(responseCode = "400", description = "Bad request. Your cancellation request was not successful.",
+              content = @Content(mediaType = "text/plain")
+          ),
+          @ApiResponse(responseCode = "422", description = "Unprocessable entity. Your cancellation request was not successful.",
+              content = @Content(mediaType = "text/plain")
+          )
+      })
+  @SecurityRequirement(name = "basic")
+  @DeleteMapping(value = "/{token}")
```

Comment:
I explicitly avoided the word `cancel` in there since this is mapped with the `DELETE` type of request. Having verbs in the API path tends to make them more rigid and is not best practice.

Reply:
Here you have `DELETE /subscription/AKERHJKWKLJJKWEJKLRJKjWER`

Reply:
Ah ja that totally makes sense